src/mds/maintain.py

```diff
@@ -59,10 +69,18 @@ async def batch_create_metadata(
 async def create_metadata(guid, data: dict, overwrite: bool = False):
     """Create metadata for the GUID."""
     created = True
+    authz = json.loads(config.DEFAULT_AUTHZ_STR)
+
+    # GUID should not be 'upload'. This will help avoid conflicts between
+    # POST /api/v1/objects/{GUID or ALIAS} and POST /api/v1/objects/upload endpoints
+    # logger.debug("checking for allowable GUIDs")
+    if guid == "upload":
+        raise HTTPException(HTTP_400_BAD_REQUEST, "GUID cannot have value: 'upload'")
```

Comment:
i think this check should also be added to `batch_create_metadata` 🤔 i'm not sure what's the best way to do that without sacrificing performance, maybe check the guid [here](https://github.com/uc-cdis/metadata-service/blob/39c26bd7dd0d6608577ac151af510803e9043d80/src/mds/maintain.py#L47), delete the record if `guid=="upload"` and return a `data_input` list?

Reply:
update: no need to delete, just add the check between these 2 lines:
```
            for data in data_list:
                if await stmt.scalar(data) == 0:
```

Reply:
Check has been added to the batch upload right after `for data in data list`.